cmd/aro/mirror.go

```diff
@@ -44,6 +48,8 @@ func mirror(ctx context.Context, log *logrus.Entry) error {
 		}
 	}
 
+	acrDomainSuffix := "." + env.Environment().ContainerRegistryDNSSuffix
```

Comment:
We should check the returned value of `azure.ContainerRegistryDNSSuffix` to make sure it is a legit DNS suffix. In [some cases](https://github.com/Azure/go-autorest/blob/f7ec17d62576b689bfc665ead655066c00c9d363/autorest/azure/environments.go#L213), the Azure SDK will return the`Not Available` literal. I am not 100% sure if we can just replace that with the `azure.PublicCloud.ContainerRegistryDNSSuffix`.

Reply:
The immediate purpose of this work is to prepare to have ARO in US Government Cloud. Adding validation for the event that German Cloud is the current cloud environment seems premature.

@jim-minter, what do you think?

Reply:
I was hoping we can just do a regex validation to determine if it's a legit DNS name or not, regardless of which cloud.

Reply:
How about adding a validator in env.NewCore that validates that the Environment is a recognised one?

Reply:
sure

Reply:
Created [8872444](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/8872444) for follow-up work